tuf/api/serialization/json.py

```diff
@@ -36,7 +36,7 @@ def deserialize(self, raw_data: bytes) -> Metadata:
             metadata_obj = Metadata.from_dict(json_dict)
 
         except Exception as e:
-            raise DeserializationError from e
+            raise DeserializationError("Failed to deserialize JSON") from e
```

Comment:
Would it make sense to include the original error message? E.g.:
```suggestion
            raise DeserializationError(f"Failed to deserialize JSON: {e}") from e
```

Same comment applies below.

Reply:
The few cases that I tried did not end up very useful: the errors can be really varied and are often not informative without more context. This context is IMO only available with the stack trace and source code.

As an example if you remove "scheme" from a key, you would get this is in client example:
`Failed to download target x: Failed to deserialize JSON: 'scheme'`
there is context there (we're getting a KeyError while parsing the json dict) but this is really only apparent if you look at the stack trace...

Reply:
I suggested a related improvement to the client example in  #1862 (optionally show the stack trace, so we usually get something human readable but can do developer readable output as well).